avocado/core/mux.py

```diff
@@ -29,6 +36,152 @@
 REMOVE_VALUE = 1
 
 
+class MuxTree(object):
+
+    """
+    Object representing part of the tree from the root to leaves or another
+    multiplex domain. Recursively it creates multiplexed variants of the full
+    tree.
+    """
+
+    def __init__(self, root):
+        """
+        :param root: Root of this tree slice
+        """
+        self.pools = []
```

Comment:
Sure, that falls into category 1 (describing the reason of the changes if not obvious).